lib/ansible/module_utils/urls.py

```diff
@@ -728,7 +728,7 @@ def http_request(self, req):
                 port = proxy_parts.get('port') or 443
                 s = socket.create_connection((proxy_parts.get('hostname'), port))
                 if proxy_parts.get('scheme') == 'http':
-                    s.sendall(self.CONNECT_COMMAND % (self.hostname, self.port))
+                    s.sendall((self.CONNECT_COMMAND % (self.hostname, self.port)).encode())
```

Comment:
This should likely be using `ansible.module_utils._text.to_bytes` instead of `encode`